File: ArraysAndStrings/1.4/Python/palindrome_perm.py
def is_palindrome(string):
    temp = set()
    if len(string)<2:
        return False
    stripped = string.replace(' ', '')
    for i in string:
        if not i.isspace():
            if i.lower() not in temp:
                temp.add(i.lower())
            else:
                temp.remove(i.lower())
    if len(temp)<2:
        return True
    return False

if __name__=="__main__":
    string = input("Introduce string\n")
    is_pal = is_palindrome(string)
    if is_pal:
        print(string, "is a palindrome permutation")
    else:
        print(string, "is not a palindrome permutation")

File: ArraysAndStrings/1.4/Golang/palindrome_perm.go
package main

import (
	"fmt"
	"bufio"
	"os"
	"strings"
)

func is_palindrome(word string) bool {
	wordS := strings.TrimSpace(word)
	wordS = strings.ReplaceAll(wordS, " ", "")
	wordS = strings.ToLower(wordS)
	temp := make(map[byte]struct{})
	for i, _ := range wordS {
		if _, ok := temp[wordS[i]]; ok {
			delete(temp, wordS[i])
		} else {
			temp[wordS[i]] = struct{}{}
		}
	}
	if len(temp) > 1 {
		return false
	}
	return true
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Println("Introduce a word")
	word, _ := reader.ReadString('\n')

	is := is_palindrome(word)
	if is {
		fmt.Println("It is a palindrome!")
	} else {
		fmt.Println("It is not a palindrome :(")
	}
}